source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -1610,20 +1610,26 @@ namespace RendererQml
 		uiTimeInput->Property("selectedTextColor", "'white'");
         uiTimeInput->Property("property string selectedTime", "\"""\"");
 		uiTimeInput->Property("width", "parent.width");
-		uiTimeInput->Property("placeholderText", !input->GetPlaceholder().empty() ? input->GetPlaceholder() : "\"Select time\"");
+		uiTimeInput->Property("placeholderText", Formatter() << "\"" << (!input->GetPlaceholder().empty() ? input->GetPlaceholder() : "Select time") << "\"");
         uiTimeInput->Property("color", context->GetColor(AdaptiveCards::ForegroundColor::Default, false, false));
 
 		uiTimeInput->Property("validator", "RegExpValidator { regExp: /^(--|[01][0-9|-]|2[0-3|-]):(--|[0-5][0-9|-])$/}");
 
 		if (!input->GetValue().empty() && Utils::isValidTime(value))
 		{
 			std::string defaultTime = value;
+            std::string defaultSelectedTime = value;
 			if (is12hour == true)
 			{
 				defaultTime = Utils::defaultTimeto12hour(defaultTime);
+                defaultSelectedTime = Utils::defaultTimeto24hour(defaultSelectedTime);
 			}
+            else
+            {
+                defaultTime = defaultSelectedTime = Utils::defaultTimeto24hour(defaultTime);
+            }
 			uiTimeInput->Property("text", Formatter() << "\"" << defaultTime << "\"");
-            uiTimeInput->Property("property string selectedTime", Formatter() << "\"" << value << "\"");
+            uiTimeInput->Property("property string selectedTime", Formatter() << "\"" << defaultSelectedTime << "\"");
 		}
```

Comment:
Would this part work without ```Utils::defaultTimeto24hour()``` ?
Since``` Utils::isValidTime()``` returns true only when the input is in 24 hour format

Reply:
The issue here is `Utils::isValidTime()` will return true if time sent by json is in `HH:MM:tt` format and it displays the same on the textbox.

Reply:
Oh got it now
Thanks